cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/RecordWriter.java

```diff
@@ -139,16 +137,33 @@ public List<StreamResult> write(Iterator<Tuple2<DecoratedKey, Object[]>> sourceI
         Map<String, Object> valueMap = new HashMap<>();
         try
         {
-            Set<Range<BigInteger>> newRanges = new HashSet<>(initialTokenRangeMapping.getRangeMap().asMapOfRanges().keySet());
+            // Use list to preserve order of ranges
+            List<Range<BigInteger>> newRanges = new ArrayList<>(initialTokenRangeMapping.getRangeMap()
```

Comment:
use list to preserve the order. It is derived from keySet, so uniqueness is inherited.